src/archive.cpp

```diff
@@ -258,14 +290,26 @@ namespace librealsense
             }
         }
 
-        std::shared_ptr<metadata_parser_map> get_md_parsers() const { return _metadata_parsers; };
+        std::shared_ptr<metadata_parser_map> get_md_parsers(rs2_extension sensor_type) const override
+        {
+            if(_metadata_parsers.find(sensor_type) != _metadata_parsers.end())
+                return _metadata_parsers.at(sensor_type);
+            return nullptr;
+        };
+
+        void set_md_parsers(const rs2_extension sensor_type, const std::shared_ptr<metadata_parser_map> metadata_parsers) override
+        {
+            _metadata_parsers[sensor_type] = metadata_parsers;
```

Comment:
This may cause an issue when, for instance, Post-processing filter receives RGB and FishEye frames, both belonging to UVC_Video sensor but with different metadata types. So it will be overwritten